packages/peregrine/lib/talons/AccountInformationPage/useAccountInformationPage.js

```diff
@@ -0,0 +1,112 @@
+import { useCallback, useMemo, useState } from 'react';
+import { useMutation, useQuery } from '@apollo/client';
+import { useUserContext } from '../../context/user';
+
+export const useAccountInformationPage = props => {
+    const {
+        mutations: {
+            setCustomerInformationMutation,
+            changeCustomerPasswordMutation
+        },
+        queries: { getCustomerInformationQuery }
+    } = props;
+
+    const [{ isSignedIn }] = useUserContext();
+    const [shouldShowNewPassword, setShouldShowNewPassword] = useState(false);
+
+    const [isUpdateMode, setIsUpdateMode] = useState(false);
+
+    const { data: accountInformationData, error: loadDataError } = useQuery(
+        getCustomerInformationQuery,
+        {
+            skip: !isSignedIn,
+            fetchPolicy: 'cache-and-network',
+            nextFetchPolicy: 'cache-first'
+        }
+    );
+
+    const [
+        setCustomerInformation,
+        {
+            data: customerInformationUpdateData,
+            error: customerInformationUpdateError,
+            loading: isUpdatingCustomerInformation
+        }
+    ] = useMutation(setCustomerInformationMutation);
+
+    const [
+        changeCustomerPassword,
+        {
+            error: customerPasswordChangeError,
+            loading: isChangingCustomerPassword
+        }
+    ] = useMutation(changeCustomerPasswordMutation);
+
+    const initialValues = useMemo(() => {
+        if (customerInformationUpdateData) {
+            return {
+                customer: customerInformationUpdateData.updateCustomer.customer
```

Comment:
This should not be needed; if the mutations return the same fragment, it will automatically re-render after the cache update.